Mathlib/Combinatorics/SimpleGraph/Finpartition.lean

```diff
@@ -0,0 +1,157 @@
+/-
+Copyright (c) 2024 Jeremy Tan. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Jeremy Tan
+-/
+import Mathlib.Combinatorics.SimpleGraph.Operations
+import Mathlib.Order.Partition.Finpartition
+
+/-!
+# Dividing the edges of a graph by a vertex finpartition
+
+Consider a finpartition `P` of the vertices of a finite graph. `P` induces a coarser graph,
+whose "supervertices" are `P.parts` and whose "superedges" contain all regular edges whose vertices
+lie in the relevant parts.
+
+Then clearly taking the union over superedges of the contained edges gives back `G.edgeFinset`.
+This file proves the fact and specialises it to the case when `P` has two parts, specified by one
+of the parts. This specialisation is intended for proving Turán's theorem
+(`Mathlib.Combinatorics.SimpleGraph.Turan`).
+
+## Main statements
+
+* `G.crossEdges se`: given the superedge `se : Sym2 P.parts`, returns the finset of all edges of `G`
+  under said superedge.
+* `SimpleGraph.card_edgeFinset_eq_sum_crossEdges_card`: the superedges partition `G`'s edges.
+* `SimpleGraph.card_edgeFinset_bipartition`: special case of the above theorem for `P` comprising
+  two parts.
+-/
+
+open Finset
+
+namespace SimpleGraph
+
+variable {V : Type*} [Fintype V] [DecidableEq V] (G : SimpleGraph V) [DecidableRel G.Adj]
+  {P : Finpartition (univ : Finset V)}
+
+/-- The finset of edges whose endpoints lie in the given superedge's endpoint's parts.-/
+def crossEdges (se : Sym2 P.parts) : Finset (Sym2 V) :=
```

Comment:
We already use the name “interedges” in other parts of mathlib, so would be good to harmonise (although I don’t really care what to)